File: src/frontend/src/types/agent.ts
export type AgentStatus =
  | 'PENDING'
  | 'IN_PROGRESS'
  | 'BLOCKED'
  | 'TESTING'
  | 'COMPLETE'
  | 'FAILED'
  | 'IDLE';

export type AgentType = 'supervisor' | 'worker';

export interface Agent {
  id: string;
  name: string;
  type: AgentType;
  status: AgentStatus;
  tmux_window: string;
  session: string;
  created_at: string;
  last_activity: string | null;
  current_task: string | null;
  display_name?: string;
  role?: string;
  project_id?: string;
}

export interface AgentListResponse {
  agents: Agent[];
  total: number;
}

export interface ArchivedAgentSummary {
  id: string;
  agent_id: string;
  agent_name: string;
  agent_type: AgentType;
  archived_at: string;
}

export interface ArchivedAgent extends ArchivedAgentSummary {
  terminal_output: string | null;
}

File: src/server/models/agent.py
from pydantic import BaseModel, Field
from datetime import datetime
from enum import Enum
from typing import Optional


class AgentStatus(str, Enum):
    PENDING = "PENDING"
    IN_PROGRESS = "IN_PROGRESS"
    BLOCKED = "BLOCKED"
    TESTING = "TESTING"
    COMPLETE = "COMPLETE"
    FAILED = "FAILED"
    IDLE = "IDLE"


class AgentType(str, Enum):
    SUPERVISOR = "supervisor"
    WORKER = "worker"


class AgentRole(str, Enum):
    WORKER = "worker"
    SUPERVISOR = "supervisor"
    PROJECT_SUPERVISOR = "project-supervisor"


class Agent(BaseModel):
    id: str  # window-based ID for backward compat (e.g. "supervisor", "cmux:worker-1")
    agent_id: Optional[str] = None  # unique agent ID (e.g. "ag_0000prim", "ag_7f3k2m9p")
    name: str  # tmux window name
    display_name: Optional[str] = None  # human-readable name (defaults to name)
    type: AgentType
    role: AgentRole = AgentRole.WORKER
    status: AgentStatus = AgentStatus.IDLE
    project_id: str = "cmux"
    permanent: bool = False
    role_context: Optional[str] = None
    reset_count: int = 0
    tasks_since_reset: int = 0
    tmux_window: str
    session: str = "cmux"  # tmux session this agent belongs to
    created_at: datetime = Field(default_factory=datetime.utcnow)
    last_activity: Optional[datetime] = None
    current_task: Optional[str] = None


class AgentList(BaseModel):
    agents: list[Agent]
    total: int


class AgentMessage(BaseModel):
    content: str
    priority: int = Field(default=0, ge=0, le=10)
